packages/@aws-cdk/aws-apigateway/lib/requestvalidator.ts

```diff
@@ -0,0 +1,85 @@
+import { Construct, PhysicalName, Resource } from '@aws-cdk/cdk';
+import { CfnRequestValidator, CfnRequestValidatorProps } from './apigateway.generated';
+import { IRestApi, RestApi } from './restapi';
+
+export interface IRequestValidatorRef {
+  /** @attribute */
+  readonly requestValidatorId: string;
+}
+
+export interface RequestValidatorProps {
+  /**
+   * The rest API that this model is part of.
+   *
+   * The reason we need the RestApi object itself and not just the ID is because the model
+   * is being tracked by the top-level RestApi object for the purpose of calculating it's
+   * hash to determine the ID of the deployment. This allows us to automatically update
+   * the deployment when the model of the REST API changes.
+   *
+   * @default None: This is automatically populated by the api.addRequestValidator method
+   */
+  readonly restApi?: IRestApi;
+
+  /**
+   * The name of this request validator.
+   *
+   * @default None
+   */
+  readonly requestValidatorName?: PhysicalName;
```

Comment:
Change back to `string` and follow existing examples on how to pass in physical name to parent and use it to synthesize attributes and send to L1.

Reply:
done